common/src/main/java/dev/ftb/mods/ftbranks/impl/FTBRanksAPIImpl.java

```diff
@@ -106,42 +95,46 @@ public static InteractionResultHolder<Component> serverChat(ServerPlayer player,
 			cachedNameForChat.setStyle(cachedNameForChat.getStyle().withHoverEvent(new HoverEvent(HoverEvent.Action.SHOW_TEXT, new TextComponent(s))));
 		}
 
+
 		main.append(cachedNameForChat);
 		main.append(" ");
-		//main.appendText("<").appendSibling(event.getPlayer().getDisplayName()).appendText(">").appendText(" ");
-
-		String message = eventMessage.trim();
 
-		Component textWithLinks = TextComponentUtils.withLinks(message);
-		TextComponent text = textWithLinks instanceof TextComponent ? (TextComponent) textWithLinks : new TextComponent(message);
 
+		TranslatableComponent fullComp = (TranslatableComponent) component;
```

Comment:
Can you add safeguards here to prevent issues with other mods that override chat format, i.e. a check whether this *is* actually a TranslatableComponent and whether the key of said component is "chat.type.text"?

Reply:
Thank you for that critique! That's a solid plan, I've done as you asked, currently running it with a few other mods that format chat to test out compatibility before sending up the changes.

I do want to add that I went a bit farther in regards to Safeguarding the TranslatableComponent casting.

I added y'alls original `String message` & `Component textWithLinks` back in, then re-organized how I was going about using `TranslatableComponent fullComp` based on your motive.

Reply:
Hmm, in this new version you're still casting component to TranslatableComponent without a check, couldn't this be moved to the `if` branch (currently in L106), i.e.
```java
if(component instanceof TranslatableComponent) {
  TranslatableComponent fullComp = (TranslatableComponent) component;
  // ...
}
```

(obviously on 1.18 this could be merged into a single instanceof with assignment)

Reply:
> Hmm, in this new version you're still casting component to TranslatableComponent without a check, couldn't this be moved to the `if` branch (currently in L106), i.e.
> 
> ```java
> if(component instanceof TranslatableComponent) {
>   TranslatableComponent fullComp = (TranslatableComponent) component;
>   // ...
> }
> ```
> 
> (obviously on 1.18 this could be merged into a single instanceof with assignment)

That was something I already attempted and instead of functioning as you stated in response, it instead was never firing, causing it to always go back to the default method of how chat was handled.

If I worded that badly, what I mean to say is that if I don't make the cast it never fires, and I tested this a lot as I'm not a fan of committing anything that does not work, unless I have too.

9/10 times the casting never errors, and this was tested with multiple random mods such as Emojifull for example.
When you brought up adding "FailSafe" parameters I took a deeper look into it and figured that it would be a safe move to put back in the code that y'all already had and simply just add a small check for what I added that way in the end, if something were to go wrong it will always fall back to what y'all already have set up.

but as far as 
`if(component instanceof TranslatableComponent) {
  TranslatableComponent fullComp = (TranslatableComponent) component;
  // ...
}`
I kept testing it with both Quark and Emojifull and it was never getting fired on my end as expected.

Reply:
> but as far as
`if(component instanceof TranslatableComponent) { TranslatableComponent fullComp = (TranslatableComponent) component; // ... }`
I kept testing it with both Quark and Emojifull and it was never getting fired on my end as expected.

Hmm, are you sure about that? I've been cleaning up your PR a little and tested this out and it *seems* to work just fine. Do you mind if I pushed my changes so you could test them out and verify they work (I'd obviously revert them if they don't)

Reply:
Yeah I can give it another go

Reply:
I'll throw back in the if check here in a second locally, Thinking something is up with the ide, as indexing has just completely stopped. Will come back to you once I can resolve this first, haha.

Reply:
Just pushed it, should be ready in a couple of minutes!

Reply:
> Just pushed it, ~should be ready in a couple of minutes!~ (nevermind, I forgot we don't do PR snapshots, feel free to just pull and build this yourself haha)

Awesome! Yes pulling on my end is pretty quick glad you could just send that over! 
Took peak and gave it a run after dealing with ide issue, and yes everything works. I'll have to look back at what I missed then.

Thank you max!